### PR TITLE
Button fixes

### PR DESCRIPTION
**Pull request by tishoyanchev**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

- Added an internal State for icon positioning 
- Made the external Prop immutable 
- Removed the gap when no ifx-icon is rendered by adding ifx-icon:empty { display: none }
- remove unused code from the stylesheet
- added the component in the React examples folder

<details>
  <summary>📦 Published PR as canary version: <code>18.2.7--canary.199.6e0502cdbddc227f76364f7e3efbd4d00db9a735.1</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @infineon/infineon-design-system-stencil@18.2.7--canary.199.6e0502cdbddc227f76364f7e3efbd4d00db9a735.1
  # or 
  yarn add @infineon/infineon-design-system-stencil@18.2.7--canary.199.6e0502cdbddc227f76364f7e3efbd4d00db9a735.1
  ```
</details>

### CODE REVIEW

File: examples/wrapper-components/react-javascript/src/App.js [hunk 1]
```diff
@@ -1,32 +1,12 @@
 import './App.css';
-
-
-import { IfxAccordion, IfxAccordionItem } from '@infineon/infineon-design-system-react';
-
+import Button from './components/button/button';
```
Comment (verena-ifx) [inline examples/wrapper-components/react-javascript/src/App.js:3]: We do not want to import the Button from the components folder here, but use the exported IfxButton from our wrapper package

Comment (tishoyanchev) [reply examples/wrapper-components/react-javascript/src/App.js:3]: Why? 
This is the App.js file. This is the file that gets shown in the browser when you run your React app. The IfxButton component is imported and used in the Button component file in React, and then in App.js, we import that React component , and we render it. This is imply used to render the React Button.

File: examples/wrapper-components/react-javascript/src/App.js [hunk 2]
```diff
@@ -1,32 +1,12 @@
...
-        I have no idea.
-      </IfxAccordionItem>
-    </IfxAccordion>
-  )
+    <div>
+      <Button />
+    </div>
+  );
 }
-
```
[the remark this answers is not in the document]
Comment (tishoyanchev) [reply examples/wrapper-components/react-javascript/src/App.js:31]: Again, here, I am not adding the IfxButton, but the React Button component that I created with IfxButton. This is just an example React app that uses the IfxButton component.

[the remark this answers is not in the document]
Comment (verena-ifx) [reply examples/wrapper-components/react-javascript/src/App.js:31]: the ifxButton is the React button, its imported from the wrapper package

File: examples/wrapper-components/react-javascript/src/App.js (hunk 1 above)
Comment (tishoyanchev) [reply examples/wrapper-components/react-javascript/src/App.js:3]: and we do, inside the components/button folder, where I create button.js file, inside of which I import IfxButton. This is how we create a React component, and then to render it on the browser, we import it inside the App.js. You do it the same way in Vue. You have /components folder, and then you have the App.vue, where you import and render Vue components, which use our Stencil components.

File: examples/wrapper-components/react-javascript/src/App.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (tishoyanchev) [reply examples/wrapper-components/react-javascript/src/App.js:31]: The IfxButton is the Stencil component with a React wrapper. A React component is not the same as a Stencil component that has a React wrapper. If you want to test our Stencil components on React, then you need to create a React component which imports our Stencil components, which is what I've done, and what you have done in Vue. 
Once again, I am importing IfxButton inside button.js file, which in order to render on the browser, you have to import it inside the App.js, the same way you are doing in Vue. If you import the IfxButton inside the App.js, then you are not creating a new React component, you are just rendering a Stencil component on React, and as you mentioned numerous times, we want to 'add' more components to React.

[the remark this answers is not in the document]
Comment (verena-ifx) [reply examples/wrapper-components/react-javascript/src/App.js:31]: nope, the button you take is from the components package, not the compiled version for React

**Review by verena-ifx — COMMENTED** [review]
another thing i was wondering - do we want/need a custom click event? might not be neccessary, just wondering what you think about it. But every web developer should be familiar with @click, onClick, etc. methods which can be used here

**Pull request merged**